src/app/asset-page/artstor-viewer/artstor-viewer.component.ts

```diff
@@ -406,6 +406,29 @@ export class ArtstorViewerComponent implements OnInit, OnDestroy {
             this.osdViewer.destroy();
         });
 
+        this.osdViewer.addOnceHandler("tile-drawn", () => {
+            // Get all canvas elements, first one is the main canvas, second one is the navigator
+            // The canvas in the reference strip starts as the third one in the collection
+            let canvasElements = document.getElementsByClassName("openseadragon-canvas")
+
+            // Set focus to the zoom in button after the main canvas
+            canvasElements.item(0).addEventListener("keydown", (event: KeyboardEvent) => {
+                if (event.keyCode === 9) {
+                    let ZoomInElement: HTMLElement = <HTMLElement>document.getElementById(`imageButtons-${this.osdViewerId}`)
+                    ZoomInElement.focus()
+                }
+            })
+
+            // Loop through all canvas elements in the reference strip and add event listener to keydown enter for accessibility
+            for (let i = 2; i < canvasElements.length; i++) {
+                canvasElements.item(i).addEventListener("keydown", (event: KeyboardEvent) => {
+                    if (event.keyCode === 13) {
+                        this.osdViewer.goToPage(i-2)
```

Comment:
There are a couple places where we have `2` hardcoded and it looks to be very intentional but I can't seem to figure out the intent. I'll keep reading through though.

Reply:
Pretty sure it is because the we are targeting the thumbnails in the reference strip. We essentially are counting the number of openseadragon canvases (since each thumbnail in the reference strip is technically a mini OSD viewer). When we query the canvases the first one works out being the main image being displayed, and the second one is the navigator component. Anything after those two should be thumbnails.

I wonder if we will get into trouble once we start adding more viewers via the compare mode. I think that will throw off the count we assume.

Reply:
Ahh 👍 now seeing the comment on line 411. I wonder if we could use something like https://openseadragon.github.io/docs/OpenSeadragon.World.html#getItemCount could help us out?

Reply:
Ah interesting idea to get the count from openseadragon. I think we also need to setup tabbing logic for the items in the reference strip though too. I threw out another potential solution here: https://github.com/ithaka/aiw-ui/pull/1207#discussion_r363758500 . Not sure we would be able to use `getItemCount` here though since we need to setup the `tabindex` on the thumbnails anyways.